src/components/Contact.js

```diff
@@ -39,19 +48,37 @@ class Contact extends React.Component {
   }
   
   get getAppropriateJSX() {
-    const { isFavourite } = this.props;
-    const { contact } = this.state;
+    const { isFavourite, contact } = this.props;
     if (!contact) return <InLineSpinner />;
     if (isFavourite) return this.createFavouriteContactJSX;
     return this.createNormalContactJSX;
   }
 
-  changeIsCheckedStatus() {
-    this.setState((prevState) => {
-      const contact = { ...prevState.contact };
-      contact.isChecked = !contact.isChecked;
-      return { contact };
-    });
+  changeIsCheckedStatus = () => {
+    const { methods, contact } = this.props;
+    methods.changeIsChecked(contact.id);
+  }
+
+  isNumberLikesBiggerThanMax = () => {
```

Comment:
isNumberLikesBiggerThanMax and isNumberLikesLowerThanMin does not require seperate functions as those only have 1 line of code. You can do: contact.likes >= LIKES.MAX directly where you need it